frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -27,13 +27,65 @@ const selectorReducer = (state: State, action: Action): State => {
       };
     }
     case "REMOVE_PROJECTS": {
-      // TODO: also remove any upstreams or downstreams related (only) to the project.
-      // if group == Groups.PROJECT, hide exclusive downstream upstreams
-      //
-      return {
-        ...state,
-        [action.payload.group]: _.omit(state[action.payload.group], action.payload.projects),
-      };
+      const newRemoveProjectsState = { ...state };
+
+      newRemoveProjectsState[action.payload.group] = _.omit(
+        state[action.payload.group],
+        action.payload.projects
+      );
+
+      const upstreamsToRemove = [];
+      const downstreamsToRemove = [];
+
+      // remove any upstreams or downstreams exclusive to the project from Group.UPSTREAM/Group.DOWNSTREAM
+      if (action.payload.group == Group.PROJECTS) {
+        action.payload.projects.forEach(p => {
+          _.forIn(state.upstreamToProject, (v, k) => {
+            // if an upstream is associated to the project
+            if (v[p]) {
+              // if the upstream is exclusive to project
+              if (Object.keys(v).length == 1) {
+                upstreamsToRemove.push(k);
+                // remove as it was only tied to this project and b/c both the project and upstream will be removed from the state
+                // otherwise we'll end up with stale information
+                delete state.upstreamToProject[k];
```

Comment:
scenario: we remove project `foo` and upstream `bar` is tied only to `foo`, which we're going to remove. so we delete `"bar": {"foo": true}` b/c this mapping is stale since we've removed `foo` and `bar` from the state.
```
upstreamToProject: {
  "bar": {"foo": true}
}
```